### PR TITLE
Abi

### PR DESCRIPTION
**Pull request by bergauz**

### CODE REVIEW

**Comment by r-gochain** [comment]
@bergauz please apply changes that Jordan requested

File: src/app/modules/wallet/wallet-send/wallet-send.component.ts [hunk 1]
```diff
@@ -110,6 +138,47 @@ export class WalletSendComponent implements OnInit {
     }));
   }
 
+  private getContractData(addrHash: string) {
+    forkJoin(
+      this._commonService.getAddress(addrHash),
+      this._commonService.getContract(addrHash),
```
[the remark this answers is not in the document]
Comment (bergauz) [reply src/app/modules/wallet/wallet-send/wallet-send.component.ts]: @jmank88  it's fast and we are going to merge Contract to Address

File: src/app/utils/constants.ts [hunk 2]
```diff
@@ -81,7 +86,262 @@ export const TOKEN_TYPES = {
...
+    ],
+    'payable': false,
+    'stateMutability': 'view',
+    'type': 'function'
+  },
+};
+
+export const ERC_INTERFACE_IDENTIFIERS = {
+  [ErcName.Erc20]: [InterfaceName.Allowance, InterfaceName.Approve, InterfaceName.BalanceOf, InterfaceName.TotalSupply, InterfaceName.Transfer, InterfaceName.TransferFrom],
+  [ErcName.Erc721]: [InterfaceName.Approve, InterfaceName.BalanceOf, InterfaceName.GetApproved, InterfaceName.IsApprovedForAll, InterfaceName.OwnerOf, InterfaceName.SafeTransferFrom, InterfaceName.SafeTransferFrom1, InterfaceName.SetApprovalForAll, InterfaceName.TransferFrom],
```
[the remark this answers is not in the document]
Comment (bergauz) [reply src/app/utils/constants.ts]: @jmank88 yeah, maybe regular json?

File: server/models/contract.go [hunk 3]
```diff
@@ -3,15 +3,15 @@ package models
...
-	CreatedAt       time.Time `json:"created_at" bson:"created_at"`
-	UpdatedAt       time.Time `json:"updated_at" bson:"updated_at"`
+	Address         string      `json:"address" bson:"address"`
+	Bytecode        string      `json:"byte_code" bson:"byte_code"`
+	Valid           bool        `json:"valid" bson:"valid"`
+	ContractName    string      `json:"contract_name" bson:"contract_name"`
+	CompilerVersion string      `json:"compiler_version" bson:"compiler_version"`
+	Optimization    bool        `json:"optimization" bson:"optimization"`
+	SourceCode      string      `json:"source_code" bson:"source_code"`
+	Abi             interface{} `json:"abi" bson:"abi"`
```
[the remark this answers is not in the document]
Comment (bergauz) [reply server/models/contract.go]: @jmank88 it makes sense

**Pull request merged**